pkgs/servers/windmill/default.nix

```diff
@@ -0,0 +1,133 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, buildNpmPackage
+, pkg-config
+, openssl
+, python3
+, deno
+, swagger-cli
+, nsjail
+, lld
+, makeWrapper
+, rustfmt
+, go
+, bash
+, rust
+, stdenv
+, fetchurl
+}:
+
+let
+  pname = "windmill";
+  version = "1.60.0";
+
+  fullSrc = fetchFromGitHub {
+    owner = "windmill-labs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-tLvfExD4FCTrsgBHg8KmaCced5Z0CejlpIWoXM/bxzw=";
+  };
+
+  pythonEnv = python3.withPackages (ps: [ ps.pip-tools ]);
+
+  # a transitive cargo dependency requires nightly rustfmt
+  # https://github.com/oxidecomputer/progenitor/blob/a90b5284979a2c00457bfd573229908a2a3bf25d/progenitor-impl/src/lib.rs#LL439C16-L439C16
+  rustfmt-nightly = rustfmt.override { asNightly = true; };
+
+  frontend-build = buildNpmPackage {
+    inherit version;
+
+    pname = "windmill-ui";
+    src = fullSrc;
+
+    sourceRoot = "source/frontend";
+
+    npmDepsHash = "sha256-J9S/Ylxx71ZUYt9uZe9hsgzN4TX7HkpGkR/ejZJ4WMA=";
+
+    preBuild = ''
+      npm run generate-backend-client
+    '';
+
+    installPhase = ''
+      mkdir -p $out/share
+      mv build $out/share/windmill-frontend
+    '';
+  };
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version;
+  src = "${fullSrc}/backend";
+
+  SQLX_OFFLINE = "true";
+  RUSTY_V8_ARCHIVE =
+    let
+      arch = rust.toRustTarget stdenv.hostPlatform;
+      fetch_librusty_v8 = args: fetchurl {
+        name = "librusty_v8-${args.version}";
+        url = "https://github.com/denoland/rusty_v8/releases/download/v${args.version}/librusty_v8_release_${arch}.a";
+        sha256 = args.shas.${stdenv.hostPlatform.system};
+        meta = { inherit (args) version; };
+      };
+    in
+    fetch_librusty_v8 {
+      version = "0.49.0";
+      shas = {
+        x86_64-linux = "sha256-l6+NdMzYI9r2aHU7/OUhbgmc/LmAZjEFL8y8GrJ+EX8=";
+        aarch64-linux = "sha256-uo+/Wsrlkm+xotoIr8xlQWoiWzMz02TKFW+olfXtpz8=";
+        x86_64-darwin = "sha256-gWxljTgt6aXUzwex2zu46B9YzTvhN0Pi9C1Ll1eiohg=";
+        aarch64-darwin = "sha256-/UnBIQ/wA/0biIG9vIDKylhqFJ8QCoqjKH7xiePZ/eg=";
+      };
+    };
```

Comment:
This is missing an or throw unsupported platform on the attr otherwise it won't eval on other platforms.

Reply:
Shouldn't the `meta.platforms` field suffice?

Reply:
@SuperSandro2000

Reply:
meta.platforms is only about building. The code should still evaluate on other platforms.